BOJ/1600.말이 되고픈 원숭이/6047198844.cpp

```diff
@@ -0,0 +1,79 @@
+#include <iostream>
+#include <queue>
+
+using namespace std;
+
+int board[200][200][31];
+
+int dy[4] = {0,0,-1,1};
+int dx[4] = {-1,1,0,0};
+int horse_dy[8] = {-1,-2,-2,-1,1,2, 2, 1};
+int horse_dx[8] = {-2,-1, 1, 2,2,1,-1,-2};
+bool range(int W,int H, int y, int x) {
+	if (0 > y || y >= H)
+		return false;
+	if (0 > x || x >= W)
+		return false;
+	return true;
+}
+
+int bfs(int W,int H,int K) {
+	if (W == 1 && H == 1)
+		return 0;
+	pair <pair<int, int>, int> start = { {0,0},K };
+	queue < pair <pair<int, int>, int>> q;
+	q.push(start);
+	for(int i=0;i<=K;i++)
+		board[0][0][i] = 1;
+	int res = 0;
+	while (!q.empty()) {
+		int cnt = q.size();
+		while (cnt--) {
+			int y = q.front().first.first;
+			int x = q.front().first.second;
+			int k = q.front().second;
+			q.pop();
+			if (y == H - 1 && x == W - 1)
+				return res;
+			//�����¿�
+			for (int i = 0; i < 4; i++) {
+				int check_y = y + dy[i];
+				int check_x = x + dx[i];
+				if (range(W, H, check_y, check_x) && !board[check_y][check_x][k]) {
+					q.push({ { check_y,check_x }, k });
+					board[check_y][check_x][k] = 1;
+				}
+			}
+
+			//���� �����ϼ��ֳ� �Ǵ�
+			if (k) {
+				--k;
```

Comment:
```suggestion
			if (k--) {
			
```
위에서는 `while (cnt--)` 라고 작성하셔서 짚어보았습니다. 사실 명확한 의미 전달을 위해서는 분리해서 표기하는 것이 바람직합니다. 코테에서는 신경 안써도 되는 부분입니다 ㅋㅋ